.github/workflows/dependabot_pr.yml

```diff
@@ -22,6 +22,33 @@ jobs:
         with:
           token: ${{ steps.github_app_token.outputs.token }}
 
+
+      # Rebase Strategy
+      - name: Rebase PR branch onto base branch
+        run: |
+          git fetch origin
+          git checkout ${{ github.head_ref }}
+          git rebase origin/${{ github.base_ref }} || git rebase --abort
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # Resolving merge conflicts
+      - name: Attempt to resolve conflicts
+        run: |
+          git merge --strategy-option theirs || git merge --abort
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Commit the changes after resolving conflicts
+        if: success()
+        uses: stefanzweifel/git-auto-commit-action@v5
```

Comment:
I'm unsure about using github actions hosted on private repos.
@peternied - Can you share what is the recommendation on this?

Reply:
I'm not sure what the question is? This workflow is only triggered by dependabot creates a pull request, if a fork has a dependabot setup it will execute there just like on opensearch-project.